src/extension/ai/stream.ts

```diff
@@ -82,45 +82,45 @@ export class StreamCreator {
     }
 
     log.info('handleEvent: building request')
-    let req = this.handlers.buildRequest(event, firstRequest)
+    this.handlers.buildRequest(event, firstRequest).then((req) => {
```

Comment:
You could likely make `handleEvent = async (event: ...` and use await here instead of `then`. Not an issue but more readable.

Reply:
> You could likely make handleEvent = async (event: ... and use await here instead of then. Not an issue but more readable.

I'll take a look at making that change. I thought about making the function async and I wasn't sure how deep that would propogate if I had to make the callers async but I'll give it a shot.